src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -340,20 +340,28 @@ void RegisterAppInterfaceRequest::Run() {
         msg_params.getElement(strings::app_hmi_type);
 
     for (size_t i = 0; i < app_type.length(); ++i) {
-      if (mobile_apis::AppHMIType::NAVIGATION ==
+      mobile_apis::AppHMIType::eType current_app_type =
           static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
-        application->set_is_navi(true);
-      }
-      if (mobile_apis::AppHMIType::COMMUNICATION ==
-          static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
-        application->set_voice_communication_supported(true);
-      }
-      if (mobile_apis::AppHMIType::PROJECTION ==
-          static_cast<mobile_apis::AppHMIType::eType>(
-              app_type.getElement(i).asUInt())) {
-        application->set_mobile_projection_enabled(true);
+              app_type.getElement(i).asUInt());
+
+      switch (current_app_type) {
+        case mobile_apis::AppHMIType::NAVIGATION: {
+          application->set_is_navi(true);
+          break;
```

Comment:
why have a break in each of the case?
AppHMIType is an array, an application can have multiple types at the same time, for example an app can be both a projection and remote_control type.

Reply:
@yang1070 you are right, AppHMIType it is an array and we check **all the values of this array** in the for cycle.
Fox example if app projection and remote_control - on the first loop of the cycle, we get 
*current_app_type = projection -> go to the case 
mobile_apis::AppHMIType::PROJECTION and set_mobile_projection_enabled(true);*
and then we don't need to go into other case blocks, 
because on the second loop of the cycle we get 
*current_app_type = remote_control -> go to the case 
mobile_apis::AppHMIType::REMOTE_CONTROL and set_remote_control_supported(true);*

keyword "break" here  - breaks out of the switch block, but not out of the for cycle

Reply:
@LitvinenkoIra you are correct.  I misunderstood this. thank you